### PR TITLE
Migrate generated bio management to use indirection with Repository and Adapter patterns

### PR DESCRIPTION
**Pull request by mark-torres10**
## Summary by CodeRabbit

* **New Features**
  * Profile bios now include creation timestamps
  * Improved generated-bio persistence and retrieval for more reliable bios

* **User-facing Improvements**
  * Viewing tool now reports basic generated-bio statistics (counts and previews)

* **Tests**
  * Added comprehensive unit and integration tests to ensure bio storage, retrieval, and edge-case handling

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: jobs/generate_profile_bios.py [hunk 1]
```diff
@@ -133,13 +136,18 @@ def main():
...
-        write_generated_bio_to_database(profile.handle, generated_bio)
-        print(f"Generated bio for {profile.handle}: {generated_bio}")
+        generated_bio_text: str = generate_bio_for_profile(profile, posts)
+        generated_bio = GeneratedBio(
+            handle=profile.handle,
+            generated_bio=generated_bio_text,
+            created_at=get_current_timestamp(),
+        )
+        generated_bio_repo.create_or_update_generated_bio(generated_bio)
+        print(f"Generated bio for {profile.handle}: {generated_bio_text}")
```
[the remark this answers is not in the document]
Comment (mark-torres10) [reply jobs/generate_profile_bios.py]: we're OK with this, we want this error to happen as that's unexpected behavior and undesired.
